ovirt/resource_ovirt_image_transfer.go

```diff
@@ -0,0 +1,367 @@
+		log.Printf("OPTIONS call on %s: %v", hostUrl.String(), res.StatusCode)
+		if err == nil && res.StatusCode == 200 {
+			return hostUrl.String(), nil
+		}
+		// can't reach the host url, try the proxy.
+	}
+
+	proxyUrl, err := url.Parse(transfer.MustProxyUrl())
+	if err != nil {
+		log.Printf("failed to parse the proxy url (%s) : %s", transfer.MustProxyUrl(), err)
+		return "", err
+	}
+	optionsReq, err := http.NewRequest(http.MethodOptions, proxyUrl.String(), strings.NewReader(""))
+	res, err := insecureClient.Do(optionsReq)
+	log.Printf("OPTIONS call on %s: %v", proxyUrl.String(), res.StatusCode)
+	if err == nil && res.StatusCode == 200 {
+		return proxyUrl.String(), nil
+	}
+	return "", err
+}
+
+func resourceOvirtImageTransferRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*ovirtsdk4.Connection)
+	getDiskResp, err := conn.SystemService().DisksService().
+		DiskService(d.Id()).Get().Send()
+	if err != nil {
+		return err
+	}
+
+	disk, ok := getDiskResp.Disk()
+	if !ok {
+		d.SetId("")
+		return nil
+	}
+	d.Set("name", disk.MustAlias())
+	d.Set("size", disk.MustProvisionedSize()/int64(math.Pow(2, 30)))
+	d.Set("format", disk.MustFormat())
+
+	if sds, ok := disk.StorageDomains(); ok {
+		if len(sds.Slice()) > 0 {
+			d.Set("storage_domain_id", sds.Slice()[0].MustId())
+		}
+	}
+	if alias, ok := disk.Alias(); ok {
+		d.Set("alias", alias)
+	}
+	if shareable, ok := disk.Shareable(); ok {
+		d.Set("shareable", shareable)
+	}
+	if sparse, ok := disk.Sparse(); ok {
+		d.Set("sparse", sparse)
+	}
+
+	return nil
+}
+
+func resourceOvirtImageTransferDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*ovirtsdk4.Connection)
+	diskService := conn.SystemService().
+		DisksService().
+		DiskService(d.Id())
+
+	return resource.Retry(2*time.Minute, func() *resource.RetryError {
+		log.Printf("[DEBUG] Now to remove Disk (%s)", d.Id())
+		_, e := diskService.Remove().Send()
+		if e != nil {
+			if _, ok := e.(*ovirtsdk4.NotFoundError); ok {
+				log.Printf("[DEBUG] Disk (%s) has been removed", d.Id())
+				return nil
+			}
+			return resource.RetryableError(fmt.Errorf("Error removing Disk (%s): %s", d.Id(), e))
+		}
+		return resource.RetryableError(fmt.Errorf("Disk (%s) is still being removed", d.Id()))
+	})
+}
+
+// ImageTransferStateRefreshFunc returns a resource.StateRefreshFunc that is used to watch
+// an oVirt image transfer.
+func ImageTransferStateRefreshFunc(conn *ovirtsdk4.Connection, transferId string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		r, err := conn.SystemService().
+			ImageTransfersService().
+			ImageTransferService(transferId).
+			Get().
+			Send()
+		if err != nil {
+			if _, ok := err.(*ovirtsdk4.NotFoundError); ok {
+				// should occur only if the transfer was deleted
+				return nil, "", nil
+			}
+			return nil, "", err
+		}
+
+		return r.MustImageTransfer(), string(r.MustImageTransfer().MustPhase()), nil
+	}
+}
+
```

Comment:
I'll add a reference for that header parsing

Reply:
Hi rgolangh,

To import the image with terraform. The images are pending. When this happens, it is usually due to the certificate.

I did the test by passing the CA certificate and without passing the certificate. Either way, the disk is pending to be imported into the data domain.

Is there any way to test this problem?

Thanks.